Control/Applicative/Permutations.hs

```diff
@@ -66,6 +68,12 @@ instance Alternative m => Applicative (Permutation m) where
     where
       lhsAlt = (<*> rhs) <$> v
       rhsAlt = (lhs <*>) <$> w
+#if MIN_VERSION_base(4,10,0)
+  liftA2 f lhs@(P x v) rhs@(P y w) = P (liftA2 f x y) (lhsAlt <|> rhsAlt)
```

Comment:
Why this one is necessary?

Reply:
`liftA2` may be more efficient than `<$>` with `<*>` for the underlying functor, in which case we want to take advantage of that.

Reply:
I think that this is correct. There are cases where laziness can be improved with `liftA2`.